src/main/java/com/redhat/quarkus/mandrel/collector/report/endpoints/ImageStatsResource.java

```diff
@@ -79,6 +83,63 @@ public ImageStats[] getByTag(@PathParam("tag") String tag) {
+        nums.put("code_area_bytes", Json.createArrayBuilder());
+        nums.put("image_total_bytes", Json.createArrayBuilder());
+        nums.put("image_heap_bytes", Json.createArrayBuilder());
+        nums.put("resources_bytes", Json.createArrayBuilder());
+        nums.put("resources_count", Json.createArrayBuilder());
+        nums.put("methods_total", Json.createArrayBuilder());
+        nums.put("methods_reflection", Json.createArrayBuilder());
+        nums.put("methods_jni", Json.createArrayBuilder());
+        nums.put("methods_reachable", Json.createArrayBuilder());
+        nums.put("classes_total", Json.createArrayBuilder());
+        nums.put("classes_reflection", Json.createArrayBuilder());
+        nums.put("classes_jni", Json.createArrayBuilder());
+        nums.put("classes_reachable", Json.createArrayBuilder());
+        nums.put("fields_total", Json.createArrayBuilder());
+        nums.put("fields_reflection", Json.createArrayBuilder());
+        nums.put("fields_jni", Json.createArrayBuilder());
+        nums.put("fields_reachable", Json.createArrayBuilder());
+        for (ImageStats i : s) {
+            nums.get("tag").add(i.getTag());
+            nums.get("peak_rss_bytes").add(i.getResourceStats().getPeakRSSBytes());
+            nums.get("gc_total_ms").add((long) (i.getResourceStats().getGcTimeSeconds() * 1000));
+            nums.get("total_build_time_ms").add((long) (i.getResourceStats().getTotalTimeSeconds() * 1000));
+            nums.get("code_area_bytes").add(i.getSizeStats().getCodeCacheSize());
+            nums.get("image_total_bytes").add(i.getSizeStats().getTotalSize());
+            nums.get("image_heap_bytes").add(i.getSizeStats().getHeapSize());
+            nums.get("resources_bytes").add(i.getSizeStats().getResourcesSize());
+            nums.get("resources_count").add(i.getSizeStats().getResourcesCount());
+            nums.get("methods_total").add(i.getTotalStats().getNumMethods());
+            nums.get("methods_reflection").add(i.getReflectionStats().getNumMethods());
+            nums.get("methods_jni").add(i.getJniStats().getNumMethods());
+            nums.get("methods_reachable").add(i.getReachableStats().getNumMethods());
+            nums.get("classes_total").add(i.getTotalStats().getNumClasses());
+            nums.get("classes_reflection").add(i.getReflectionStats().getNumClasses());
+            nums.get("classes_jni").add(i.getJniStats().getNumClasses());
+            nums.get("classes_reachable").add(i.getReachableStats().getNumClasses());
+            nums.get("fields_total").add(i.getTotalStats().getNumFields());
+            nums.get("fields_reflection").add(i.getReflectionStats().getNumFields());
+            nums.get("fields_jni").add(i.getJniStats().getNumFields());
+            nums.get("fields_reachable").add(i.getReachableStats().getNumFields());
+        }
+        final JsonObjectBuilder rb = Json.createObjectBuilder();
```

Comment:
Eventually, we should model this with Java classes. This looks pretty ugly.

Reply:
I agree. I merely want to "script", for a lack of a better word, prototype to play with I can comfortable consume and plot and tie triggers to. I would like to finally cross the gap between collecting the data and having a simple, meaningful insight into "something is wrong".

I will refactor it into a POJO that could also be e.g. cached etc., later...

Reply:
At some point I would like to generalize it so as you can "start an experiment", e.g. with a specific Quarkus (or other) app, get Jenkins collect a whole matrix of platforms and also flags, settings etc., plot the whole result and terminate the experiment with some conclusion.